src/Mocking.jl

```diff
@@ -59,12 +59,12 @@ macro patch(expr::Expr)
         string(target_name)
     end
 
-    # Need to evaluate the body of the function in the context of the `@patch` macro in
-    # order to support closures.
-    # func = Expr(:(->), Expr(:tuple, params...), body)
-    func = Expr(:(=), Expr(:call, gensym(patch_name), params...), body)
+    def[:name] = gensym(string(patch_name, "_patch"))
```

Comment:
Rather than the above stuff to normalize the different forms a `target_name` might have depending on if it is a `Symbol` or an `Expr`
we can just do:

`def[:name] = gensym(Symbol(target_name, :_patch))`

I think that has literally the exact same effect.

demo
```
julia> gensym(Symbol(:foo, :_patch))
Symbol("##foo_patch#400")

julia> gensym(Symbol(:(Base.foo), :_patch))
Symbol("##Base.foo_patch#401")
```

Reply:
Good point. The normalizing is just a nicety but it's probably not work the extra complexity.

Reply:
I think the normalizing even ends up like this? No?

Reply:
I should be able to drop the extra string calls like you suggested. I was also thinking you were suggesting dropping normalization all together (e.g. `:(Base.foo)` → `:foo`). Reading your original comment again I don't think you were suggesting that but I may still do that as the main point to adjusting the gensym'd name is to improve readability in the stack traces.

Reply:
the current code doesn't normalize `:(Base.foo)` → `:foo)`
AFAICT
All it does is call `string(:(Base.foo))` which just gives `"Base.foo"` ?

Reply:
This is the normalization code:
```julia
patch_name = if target_name isa Symbol  # f(...)
    target_name
elseif target_name.head === :.  # Base.f(...)
    target_name.args[2].value
else
    string(target_name)
end
```

Reply:
I refactored the normalization code and the patch name creation. Let me know what you think.

Reply:
Oh I some how skimmed over the  `elseif` bit.
New code is fine.